poly1305/tests/lib.rs

```diff
@@ -23,7 +23,7 @@ fn test_nacl_vector() {
 
     let expected = hex!("f3ffc7703f9400e52a7dfb4b3d3305d9");
 
-    let result1 = Poly1305::new(key.as_ref().into()).compute_unpadded(&msg);
+    let result1 = Poly1305::new((&key).into()).compute_unpadded(&msg);
```

Comment:
All of those are changes not required with a bump of `hybrid-array` to `v0.2.0-rc.6`

Reply:
No, the `AsRef` in v0.2.0-rc.6 isn't this one.

Reply:
`key.as_ref()` should work I would think

Reply:
I'm not actually sure.
the as_ref would be this one: https://doc.rust-lang.org/std/primitive.array.html#impl-AsRef%3C%5BT%5D%3E-for-%5BT;+N%5D

which would produce a `&[u8]` and we definitely not be accepted with `Array::from`. 

I don't know what kind of type rustc was using before. I can't find a single one that would match that path.

Reply:
`hybrid-array` defines its own `AsRef` impl to do that conversion: https://docs.rs/hybrid-array/0.2.0-rc.7/hybrid_array/struct.Array.html#impl-AsRef%3CArray%3CT,+U%3E%3E-for-%5BT;+N%5D

Reply:
That was working before because `key.as_ref()` would evaluate as `&[u8]` and generic-array 0.14 had a `From<&[u8]>`
https://docs.rs/generic-array/0.14.7/generic_array/struct.GenericArray.html#impl-From%3C%26'a+%5BT%5D%3E-for-%26'a+GenericArray%3CT,+N%3E

(with a panic as a caveat).

Reply:
Yeah but here I have the key coming from `hex!()` which gives me a `[u8;32]`, for which I can use `Array;;from(key: &[u8; 32])`.

Anyway, that explains that.

Reply:
The `AsRef` impl I linked is defined for `[T; N]` and will work fine for a `[u8; 32]`.

Reply:
but I want to go the other way?!

Reply:
It's defined the other way as well: https://docs.rs/hybrid-array/0.2.0-rc.7/hybrid_array/struct.Array.html#impl-AsRef%3C%5BT;+N%5D%3E-for-Array%3CT,+U%3E

Have you just tried it?

Reply:
no because I was hanged up on the `From` conversion. I completely missed your point. My bad.